zerocopy-derive/src/lib.rs

```diff
@@ -1000,11 +956,6 @@ const STRUCT_UNION_UNALIGNED_CFG: Config<StructRepr> = Config {
 // - `repr(u8)` or `repr(i8)`
 
 fn derive_unaligned_enum(ast: &DeriveInput, enm: &DataEnum) -> proc_macro2::TokenStream {
-    if !enm.is_fieldless() {
-        return Error::new_spanned(ast, "only field-less enums can implement Unaligned")
-            .to_compile_error();
-    }
-
```

Comment:
Since we're removing this restriction, remove or update the comment at the bottom of this function, which currently assumes field-less enums?